zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/FastLeaderElection.java

```diff
@@ -952,11 +952,10 @@ public Vote lookForLeader() throws InterruptedException {
                  * Otherwise processes new notification.
                  */
                 if (n == null) {
-                    if (manager.haveDelivered()) {
```

Comment:
I think the original behavior is correct, in case we haven't delivered, we'll try to connect and deliver again, then waiting for new notifications to come in before timed out and send another round of notification.